api/src/test/java/org/openmrs/patient/impl/VerhoeffIdentifierValidatorTest.java

```diff
@@ -33,87 +33,138 @@
+	public void isValid_shouldFailWithIncorrectlyCalculatedSuffixes() {
+		// Test allowed identifiers that just have the wrong check digit.
+		for (String allowedIdentifier : allowedIdentifiers) {
+			assertFalse(validator.isValid(allowedIdentifier + "-" + unusedCheckDigit));
+		}
+	}
+
+	@Test
+	public void isValid_shouldFailWithNumericSuffixes() {
 		for (int i = 0; i < allowedIdentifiers.length; i++) {
 			try {
-				validator.isValid(allowedIdentifiers[i] + "-X");
-				fail("Identifier " + allowedIdentifiers[i] + " should have failed.");
-			}
-			catch (Exception e) {}
-			try {
-				validator.isValid(allowedIdentifiers[i] + "-10");
+				validator.isValid(allowedIdentifiers[i] + "-" + allowedIdentifiersCheckDigitsInt[i]);
 				fail("Identifier " + allowedIdentifiers[i] + " should have failed.");
 			}
-			catch (Exception e) {}
+			catch (Exception e) { /* Expected */ }
 		}
-		
-		//Make sure check digit can't be numeric
-		for (int j = 0; j < invalidIdentifiers.length; j++) {
-			try {
-				validator.isValid(allowedIdentifiers[j] + "-" + allowedIdentifiersCheckDigitsInt[j]);
-				fail("Identifier " + allowedIdentifiers[j] + " should have failed.");
+	}
+
+	@Test
+	public void checkDigit_shouldChangeWhenAdjacentCharsAreTransposed() {
+		VerhoeffIdentifierValidator v = new VerhoeffIdentifierValidator();
+		String test;
+		int checkDigit;
+		String pre;
+		String post;
+		int failures = 0;
+		StringBuilder failureMsg = new StringBuilder();
+		char c;
+		char d;
+		final String lineSeparator = System.getProperty("line.separator");
+		for (String allowedIdentifier: allowedIdentifiers) {
+			for (int i = 0; i < allowedIdentifier.length(); i++) {
+				checkDigit = v.getCheckDigit(allowedIdentifier);
+				for (int j = 1; j < allowedIdentifier.length(); j++) {
+					c = allowedIdentifier.charAt(j - 1);
+					d = allowedIdentifier.charAt(j);
+					// If 2 characters are not identical, shuffle them and then test that check digit is different.
+					if (c != d) {
+						pre = allowedIdentifier.substring(0, j - 1);
+						post = allowedIdentifier.substring(j + 1);
+						test = pre + d + c + post;
+						if (checkDigit == v.getCheckDigit(test)) {
+							failureMsg.append("Check digits for '");
+							failureMsg.append(allowedIdentifier);
+							failureMsg.append("' and '");
+							failureMsg.append(test);
+							failureMsg.append("' should be different, both were ");
+							failureMsg.append(checkDigit);
+							failureMsg.append(lineSeparator);
+							failures++;
+						}
+					}
+				}
+				// Shuffle the allowed identifier string around so that first character becomes last.
+				allowedIdentifier = allowedIdentifier.substring(1, allowedIdentifier.length()) + allowedIdentifier.charAt(0);
 			}
-			catch (Exception e) {}
-		}
-		
-		//Now test allowed identifiers that just have the wrong check digit.
-		for (int i = 0; i < allowedIdentifiers.length; i++) {
-			assertFalse(validator.isValid(allowedIdentifiers[i] + "-" + unusedCheckDigit));
```

Comment:
Did you intentionally get rid of these?

Reply:
This is covered in the test  isValid_shouldFailWithIncorrectlyCalculatedSuffixes()